packages/payload/src/uploads/getExternalFile.ts

```diff
@@ -18,15 +21,37 @@ export const getExternalFile = async ({ data, req, uploadConfig }: Args): Promis
       fileURL = `${baseUrl}${url}`
     }
 
-    const headers = uploadConfig.externalFileHeaderFilter
-      ? uploadConfig.externalFileHeaderFilter(Object.fromEntries(new Headers(req.headers)))
-      : { cookie: req.headers.get('cookie')! }
+    let res
+    try {
+      const headers = uploadConfig.externalFileHeaderFilter
+        ? uploadConfig.externalFileHeaderFilter(Object.fromEntries(new Headers(req.headers)))
+        : { cookie: req.headers.get('cookie')! }
+      const fetchOptions: RequestInit = {
+        credentials: 'include',
+        headers,
+        method: 'GET',
+      }
+      const allowList = uploadConfig.pasteURL ? uploadConfig.pasteURL.allowList : []
 
-    const res = await fetch(fileURL, {
-      credentials: 'include',
-      headers,
-      method: 'GET',
-    })
+      // Use native Fetch if allowList is populated
+      if (allowList.length) {
```

Comment:
The presumption here is that the fileURL is in the list (if it got this far then it should be).

I can also add a check using:
`import { isURLAllowed } from '../utilities/isURLAllowed.js'`
But I think it's redundant.